src/Modal.jsx

```diff
@@ -60,19 +87,27 @@ class Modal extends React.Component {
 			'border--none'
 		);
 
+		const overlayShim = (
+			<div className='overlayShim' onClick={this.onDismiss}>
+				<div className='inverted'></div>
+			</div>
+		);
+
 		return (
 			<div
 				role='dialog'
 				tabIndex='0'
 				onKeyDown={this.onKeyDown}
 				className={classNames}
-				{...other}>
+				{...other}
+			>
 
-				<div className='overlayShim' onClick={this.onDismiss}>
-					<div className='inverted'></div>
-				</div>
+				{!fullscreen && overlayShim}
 
-				<div className={modalClasses} >
+				<div
+					className={modalClasses}
+					style={{marginTop: this.state.topPosition}}
+				>
 					<div className='align--right'>
```

Comment:
Unrelated to this commit, but can we remove the "align--right" class? The close button should be aligned left.